src/testers.jl

```diff
@@ -182,7 +182,7 @@ function test_rrule(
         @test ∂self === NO_FIELDS  # No internal fields
 
         # Correctness testing via finite differencing.
-        x̄s_is_dne = accumulated_x̄ .== nothing
+        x̄s_is_dne = isa.(accumulated_x̄, Union{Nothing, DoesNotExist})
```

Comment:
this should have a todo.
And maybe a `Base.depwarn`?

Reply:
`depwarn` for what? we kind of use both unfortunately

Reply:
Oh, I thought this was going to close 
https://github.com/JuliaDiff/ChainRulesTestUtils.jl/issues/113
?
so we would in future just use DoesNotExist ?

Reply:
So we should `Base.Depwarn` if `nothing` is passed

Reply:
Yeah actually I think you are right, but I didn't think about this from the start so need to check.

Reply:
If i am right then we should have a deprecation warning?

Reply:
I've added one below, or am I missing something obvious?

Reply:
@oxinabox not sure if you are happy with the current depwarn?

Reply:
Cool